cmd/minishift/cmd/artifact_preflight.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright (C) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+
+	configCmd "github.com/minishift/minishift/cmd/minishift/cmd/config"
+	"github.com/minishift/minishift/pkg/minishift/oc"
+	"github.com/minishift/minishift/pkg/util"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// preflightChecksForArtifacts is executed once artifacts are cached.
```

Comment:
Do you mean `.. is executed once OC is cached`?

Reply:
@budhrg OC is one artifact, ISO is another and if something else in future we need to download then it will part of artifact.